flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/FlinkCatalog.java

```diff
@@ -229,6 +229,20 @@ public void dropDatabase(String name, boolean ignoreIfNotExists, boolean cascade
       throws DatabaseNotExistException, DatabaseNotEmptyException, CatalogException {
     if (asNamespaceCatalog != null) {
       try {
+        if (cascade) {
+          List<ObjectPath> deleteTablePaths = icebergCatalog.listTables(toNamespace(name)).stream()
```

Comment:
Consider parallel operations using `Tasks` here?

Reply:
If droping a database and droping a table under it happen at the same time, it will make no influence.
If droping a database and creating a table under it happen at the same time, it will fail, and we just need to retry it.

Reply:
I mean, Can we delete tables in parallel via `org.apache.iceberg.util.Tasks` ?